File: src/01-easy/horse-racing-duals/solution.js
const n = parseInt(readline());
const h = [];
for (let i = 0; i < n; i++) h.push(parseInt(readline()));
h.sort((a, b) => a - b);
let min = h[1] - h[0];
for (let i = 1; i < n - 1; i++) {
    const d = h[i + 1] - h[i];
    if (d < min) min = d;
}
console.log(min);

File: src/01-easy/horse-racing-duals/solution.rs
use std::io;
fn main() {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    let n: usize = s.trim().parse().unwrap();
    let mut h = Vec::with_capacity(n);
    for _ in 0..n {
        let mut s = String::new();
        io::stdin().read_line(&mut s).unwrap();
        h.push(s.trim().parse::<i64>().unwrap());
    }
    h.sort();
    let min = h.windows(2).map(|w| w[1] - w[0]).min().unwrap();
    println!("{}", min);
}
